Login/logout, Signup and Get User Endpoints

vinob09 opened the pull request:
Completed functionality for all of the following endpoints:

- Login a User
- Logout a User
- Signup a User
- Get the current User

I also added the expected error messages for some of these and cleaned up a few things. Hope this is all okay so far! I know there will be more updates on it to come 😄

philling83 reviewed (changes requested):
Adjustment to error handling for signup route. I believe if either one of the fields don't exist, you are still sending all 4 error messages rather than whichever one is missing

vinob09 commented:
Hey Philip! I went over Phase 5 stuff with express-validations so I added those instead of my custom error handling you mentioned above, both for login and signup. I also updated the err.stack on these validations and error handlers to check the err.stack condition if it is not equal to production, to show the err stack but null if it is. So everything committed should be Phase 4 & 5. Let me know if it all looks good to proceed with deployment!

philling83 reviewed (approved):
Looks good 👍🏽
